### PR TITLE
Correct and clarify Python API documentation

### PR DESCRIPTION
**Pull request by ghost**
- Address ML-4612.
- Examples of pulling metrics from historical runs waiting on #368.
- Includes theme change to color links to API entities.

### CODE REVIEW

File: docs/source/cli.rst [hunk 1]
```diff
@@ -29,13 +29,17 @@ and list experiments, and download artifacts.
...
 
+.. contents:: Table of Contents
+  :local:
+  :depth: 1
 
-Azure ML
---------
 
-Subcommands to serve models on Azure ML.
+Azure Machine Learning Models
```
Comment (aarondav) [inline docs/source/cli.rst]: Not in love with how this looks, but it's not a regression:
image

File: docs/source/models.rst [hunk 2]
```diff
@@ -5,18 +5,18 @@ MLflow Models
...
 variety of downstream tools---for example, real-time serving through a REST API or batch inference
-on Apache Spark. The format defines a convention that lets you save a model in different "flavors" that can be
-understood by different downstream tools.
+on Apache Spark. The format defines a convention that lets you save a model in different "flavors" 
+that can be understood by different downstream tools.
 
 .. contents:: Table of Contents
   :local:
-  :depth: 2
+  :depth: 1
```
Comment (aarondav) [inline docs/source/models.rst]: I kind of liked the depth-2 here, it gives a good high-level view of what models we support, and concrete examples of what we're talking about:

image

**Comment by stbof** [comment]
It's to have consistency with all the models. We can pick a different
convention.

**Comment by stbof** [comment]
BTW it's spelled out to respect the product name. Does Azure use the
acronym Azure ML?

**Comment by stbof** [comment]
Ok. Can switch back.

File: docs/theme/mlflow/static/css/theme.css [hunk 3]
```diff
@@ -5192,7 +5192,7 @@ a .rst-content tt,
 a .rst-content tt,
 a .rst-content code {
     font-weight: bold;
-    color: #052247
+    color: #2980B9;
```
Comment (aarondav) [inline docs/theme/mlflow/static/css/theme.css]: @mateiz 

Before:
image

After:
image

Seems reasonable to me.

File: mlflow/entities/run_info.py [hunk 4]
```diff
@@ -43,7 +45,7 @@ def __eq__(self, other):
         return False
 
     def copy_with_overrides(self, status, end_time):
```
Comment (aarondav) [inline mlflow/entities/run_info.py]: This method should not appear in the public documentaiton.

File: mlflow/h2o.py [hunk 5]
```diff
@@ -1,4 +1,12 @@
-"""MLflow integration for H2O."""
+"""
+The ``mlflow.h2o`` module provides an API for logging and loading H2O models. This module supports
+exporting H2O models with the following flavors:
+
+H20 (native) format
+    This is the main flavor and is always produced.
```
Comment (aarondav) [inline mlflow/h2o.py]: We actually always produce both flavors. Downstream, one or the other will be picked -- for example, the pyfunc version would be used by SageMaker for deployment (as SageMaker knows how to deploy pyfuncs, but not H2O models), while the H2O format could be loaded by the user to use H2O-specific functionality.

File: mlflow/models/__init__.py [hunk 6]
```diff
@@ -1,3 +1,20 @@
...
+
+- :py:mod:`mlflow.pyfunc`
+- :py:mod:`mlflow.h2o`
+- :py:mod:`mlflow.keras`
+- :py:mod:`mlflow.pytorch`
+- :py:mod:`mlflow.sklearn`
+- :py:mod:`mlflow.spark`
+- :py:mod:`mlflow.tensorflow`
+
+For details, see `MLflow Models <../../models.html>`_.
```
Comment (aarondav) [inline mlflow/models/__init__.py]: Oops, another broken link.

File: mlflow/projects/__init__.py [hunk 7]
```diff
@@ -83,43 +85,48 @@ def run(uri, entry_point="main", version=None, parameters=None, experiment_id=No
...
     """
-    Run an MLflow project from the given URI.
+    Run an MLflow project. The project can be local or stored at a Git URI.
 
-    Supports downloading projects from Git URIs with a specified version, or copying them from
-    the file system. For Git-based projects, a commit can be specified as the ``version``.
+    You can run the project locally or remotely on a Databricks.
+
+    For information on using this method in chained workflows, see `Building Multi-Step Workflows
+    <../../projects.html#building-multi-step-workflows>`_.
```
Comment (aarondav) [inline mlflow/projects/__init__.py]: Also a broken link -- just `../`

File: mlflow/keras.py [hunk 8]
```diff
@@ -88,12 +95,13 @@ def predict(self, dataframe):
         return predicted
 
 
-def load_pyfunc(model_file):
+def load_pyfunc(path):
```
Comment (aarondav) [inline mlflow/keras.py]: Mm, this is a bug.

File: docs/source/tracking.rst [hunk 9]
```diff
@@ -304,6 +304,13 @@ The format ``sftp://pass:user@host/`` is supported for logging in. However, for
 
 When using this store, ``pysftp`` has to be installed on both client and server. Run ``pip install pysftp`` to install the required package.
 
+NFS
```
Comment (aarondav) [inline docs/source/tracking.rst]: Added this as a follow-up to a lack of documentation on how to use NFS.

File: mlflow/tracking/service.py [hunk 10]
```diff
@@ -135,24 +135,26 @@ def set_tag(self, run_id, key, value):
         tag = RunTag(key, str(value))
         self.store.set_tag(run_id, tag)
 
-    def log_artifact(self, artifact_uri, local_path, artifact_path=None):
+    def log_artifact(self, run_id, local_path, artifact_path=None):
```
Comment (aarondav) [inline mlflow/tracking/service.py]: The changes here were made in master and I think you accidentally reverted them at some point.

File: docs/theme/mlflow/layout.html [hunk 11]
```diff
@@ -115,7 +115,7 @@
   <page>
     {# SIDE NAV BAR, TOGGLES ON MOBILE #}
 
-    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
+    <nav data-toggle="wy-nav-shift" class="wy-nav-side relative">
```
Comment (aarondav) [inline docs/theme/mlflow/layout.html]: Oops, didn't mean to make this change. Next commit removes it.

**Comment by aarondav** [comment]
Merging this -- @stbof it would be great if you could take a look at this commit https://github.com/mlflow/mlflow/pull/454/commits/f171a16d1100df26ec82a40b173a2ff9d25f3bb5 and make a follow-up PR if any of the copy I introduced there needs fixing.

**Comment by stbof** [comment]
LGTM.

**Pull request merged**